DOCSP-38757: observables (#26)
mongodb/docs-scala: source/reference/observables.txt

The {+driver-short+} is an asynchronous and non-blocking driver.
By implementing the ``Observable`` model, asynchronous events become simple, composable
operations that are free from the complexity of nested callbacks.

For asynchronous operations, there are three interfaces:

- `Observable <{+api+}/org/mongodb/scala/Observable.html>`__
- `Subscription <{+api+}/org/mongodb/scala/Subscription.html>`__
- `Observer <{+api+}/org/mongodb/scala/Observer.html>`__

.. note::

   The driver is built upon the `MongoDB Reactive Streams
   driver <https://www.mongodb.com/docs/languages/java/drivers/java-rs/current/>`__
   and is an implementation of the reactive streams specification.
   ``Observable`` is an implementation of ``Publisher`` and ``Observer``
   is an implementation of ``Subscriber``.
   
   The following class naming conventions apply:
   
   - ``Observable``: custom implementation of a ``Publisher``
   - ``Observer``: custom implementation of a ``Subscriber``
   - ``Subscription``

Observable
----------

The ``Observable`` is an extended ``Publisher`` implementation, and in general
it represents a MongoDB operation which emits its results to the ``Observer``

> based on demand requested by the ``Subscription`` to the ``Observable``.

Reviewer: I believe this syntax qualifies as incorrect

```suggestion
based on a request from the ``Subscription`` to the ``Observable``.
```
